camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/methods/DslComponent.java

```diff
@@ -255,8 +271,14 @@ public ExecuteEffector(DslComponent component, String effectorName, Map<String,
             if (targetEffector.isAbsentOrNull()) {
                 throw new IllegalArgumentException("Effector " + effectorName + " not found on entity: " + targetEntity);
             }
-            return (Task<Object>) Entities.invokeEffector(targetEntity, targetEntity, targetEffector.get(), args);
+            if (null == cachedTask) {
+                cachedTask = null == argList
```

Comment:
We also need to think about the resolved value surviving rebind. We probably want to store the result somehow, as well as the task (because persistence won't cope well with tasks).